content/guides/references/release-stages.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Release Stages
+---
+
+Cypress follows Semantic Versioning so that it is clear how you may be affected
+when upgrading to a newer version of Cypress. If there is a minor version
+increment, 11.0.0 → 11.1.0 for example, you should not experience disruption in
+your usage of Cypress. If there is a major version increment, 11.1.0 → 12.0.0
+for example, you may need to make changes to continue using Cypress. For most
+scenarios this is a valid way to consider complexity when upgrading Cypress. In
+some cases, however, a feature of cypress may be in a stage of development in
+which it does not adhere to these rules. In order to better provided
+transparency of when this might be the case, we have consolidated definitions of
+different release stages and how we approach changes and communication in those
+stages.
+
+## Experimental
+
+Experimental features consist of features that we are looking to gather user
+feedback on before we commit fully to the initial implementation or the feature
+entirely. These type of features require you to opt-in to via a configuration
+value before the feature is activated. These types of features may be unstable
```

Comment:
>  These type of features require you to opt-in to via a configuration
value before the feature is activated. 

I do not think we should narrow ourselves into a box with a configuration flag. Reading the terms for `Experimental` vs `Alpha`, they sounds almost 1:1 in terms of what users can expect in breaking changes or removal of functionality. 

Also reading the definitions of `Alpha` and `Beta`, they are identical. The key differences are how changes / breaking changes are released.

Is there any way we can condense `Alpha` & `Experimental`? What are the reasons not to condense them?

There will be "beta" E2E features which will alway sneed to be "opted" in to via configuration, which isn't to different than users "opting" into using Beta component testing setups which is also configuration based.

Reply:
Actually looking closer at `Alpha` vs `Beta`, is the only different for users in confidence on if we provide migration message if we break APIs or remove functionality?

Reply:
I had this thought too... I wonder if we really need 4 stages at all? On the other hand, In other OSS projects I work on, we've often had 4 stages:

- alpha: Basically what we call experimental - you can try it, the main purpose is feedback and power users. Expect things to not work perfectly. Things are missing.
- beta: All features are present, but likely there are some bugs/unexpected behavior.
- release candidate: Similar the final release in terms of API and features - the main point of this stage is the API is set and there won't be any major breaking changes. Users can start adopting the feature with confidence.
- GA - it's ready.

I wonder if these could help mold our definitions?

Reply:
In concept and definition I'm good with the stages listed here. The main point of the stage is convey to our users the level of risk they're taking on by relying on the feature:
* experimental - High risk, this feature may disappear
* alpha - Medium risk, this feature may change drastically before becoming GA, but it'll likely get done eventually.
* beta - low risk, this is pretty much done, but might have some breaking changes here and there for the better.
* GA - lowest risk, breaking changes are rare and will be clearly conveyed. Cypress is committed to this feature for the long haul.

The other thing to keep in mind is that we don't have to progress through the stages directly, we can skip from experimental to GA if we want (cy.origin) or start with a beta if we're confident enough.

Reply:
I made some comments [here](https://github.com/cypress-io/cypress-documentation/pull/4865#pullrequestreview-1192021452) before reading this thread, but broadly I agree with these definitions.

> The other thing to keep in mind is that we don't have to progress through the stages directly, we can skip from experimental to GA if we want (cy.origin) or start with a beta if we're confident enough.

I think it's worth adding some mention of this, that it's not required for features to hit each stage.

The reason I think we should keep both alpha and experimental is that they communicate different ideas about whether something is "on the path to release". Sometimes, the outcome of an experiment will be "we learned some things, and we decided not to build the feature". 

For an alpha stage feature, my supposition is that we've decided we definitely want it to be a part of Cypress, _that_ decision is behind us, and it's just a matter of how complete/defined the implementation is whether we call it alpha or beta. Users should expect that it's safe to use/depend on in the sense that we don't expect to remove it, even though breaking changes may occur that they have to keep up with.

I expect alpha will be the least-used stage, but it makes sense for say, a new framework or bundler in CT, where we've decided to support it, but implementation will come in stages or depends on feedback. If we aren't sure we want to support it, we'd use the experimental flag to enable, so users don't discover something in the UI that's a speculative feature on our part.

Reply:
When I dug into this, I was _sure_ I could make it so alpha and experimental had exactly the same definition, but I talked myself out of it for the above reasons.